dask_cuda/benchmarks/local_cudf_shuffle.py

```diff
@@ -21,41 +24,69 @@
 )
 
 
-def shuffle_dask(df, *, noop=False):
-    result = shuffle(df, index="data", shuffle="tasks")
-    if noop:
+def shuffle_dask(df, args):
+    result = shuffle(df, index="data", shuffle="tasks", ignore_index=args.ignore_index)
+    if args.backend == "dask-noop":
         result = as_noop(result)
     t1 = perf_counter()
     wait(result.persist())
     return perf_counter() - t1
 
 
-def shuffle_explicit_comms(df):
+def shuffle_explicit_comms(df, args):
     t1 = perf_counter()
     wait(
         dask_cuda.explicit_comms.dataframe.shuffle.shuffle(
-            df, column_names="data"
+            df, column_names="data", ignore_index=args.ignore_index
         ).persist()
     )
     return perf_counter() - t1
 
 
-def bench_once(client, args, write_profile=None):
-    # Generate random Dask dataframe
-    chunksize = args.partition_size // 8  # Convert bytes to float64
-    nchunks = args.in_parts
-    totalsize = chunksize * nchunks
-    x = da.random.random((totalsize,), chunks=(chunksize,))
-    df = dask.dataframe.from_dask_array(x, columns="data").to_frame()
+def create_df(nelem, df_type):
+    if df_type == "cpu":
+        return pd.DataFrame({"data": np.random.random(nelem)})
+
+    assert df_type == "gpu"
+    import cupy  # isort:skip
+    import cudf  # isort:skip
+
+    return cudf.DataFrame({"data": cupy.random.random(nelem)})
+
+
+def create_data(
+    client: Client, args, name="balanced-df"
+) -> Tuple[int, dask.dataframe.DataFrame]:
+    """Create a dask dataframe that are distributed evenly"""
+    workers = list(client.run(lambda: 42).keys())
+    assert len(workers) > 0
+
+    chunksize = args.partition_size // np.float64().nbytes
+    # Distribute the new partitions between workers by round robin.
+    # We use `client.submit` to control the distribution exactly.
+    # TODO: support unbalanced partition distribution
+    dsk = {}
+    for i in range(args.in_parts):
+        worker = workers[i % len(workers)]  # Round robin
+        dsk[(name, i)] = client.submit(
+            create_df, chunksize, args.type, workers=[worker], pure=False
+        )
+    wait(dsk.values())
+
+    divs = [None] * (len(dsk) + 1)
+    ret = new_dd_object(dsk, name, create_df(0, "cpu"), divs).persist()
```

Comment:
Is forcing `"cpu"` intentional here or would we want it to match `args.type`?

Reply:
Good catch, fixed